Use /run instead of /var/run as default RUNDIR

chris-se opened the pull request:
I'm making this a separate pull request because this is something where I'm not sure you'll accept or reject. In Debian we try to only use /run, and consider /var/run to be legacy - and I believe most other distributions have similar policies. (/run has been available everywhere I know of for the last 5 years or so at the very least.) I therefore believe this is generic enough to be suitable for upstream, but it's your call.

gonzoleeman commented:
If you're going to change it I'd like to see it be configurable, since SUSE is not quite there yet. Can you resubmit with it being a "config" option?

chris-se committed:
Make RUNDIR (/var/run) configurable

For quite some time now /run has been established as the default
directory for runtime information in most (if not all) mainsream
distributions, with /var/run being a symlink to /run. Hence make
it configurable via a configure switch, instead of enforcing
/var/run.

Signed-off-by: Christian Seiler <christian@iwakd.de>

chris-se commented:
Ok, I've force-pushed a patch that makes it configurable. By default `/var/run` is used, but with `--with-rundir=/run` (or similar) you can override it with `./configure`. Note: now the file `include/libisns/paths.h` needs to be generated (renamed the original to `paths.h.in`), because the rundir variable needs to be replaced.

I've build-tested this and verified that the binaries have the correct settings via the `strings` command (and the generated header file is correct).
